### ISSUE TITLE
Show All Symbols returns no results

### ISSUE THREAD

**Issue by levin-ilya** [DonJayamanne/pythonVSCode#138]
Key binding - ⌘T  Show All Symbols 

I've been trying to get this to work, but it doesn't return any results no matter what I type. Is there additional configuration for this to work, or is it not supported?

**Comment by grandcat** [comment]
Same for me.
Using VS Code v1.1.1.

**Comment by DonJayamanne** [comment]
Please could you try this with just one our files in your working directory.

**Comment by rampage644** [comment]
@DonJayamanne: was it fixed? Last time i looked into source there was no `registerWorkspaceSymbolProvider` provider.

After a brief research i found that one had to preload all project files within `jedi` as it doesn't support such feature.

**Comment by levin-ilya** [comment]
I just tried it again with the latest update, still not working even with just one or two files in the working directory. 

@rampage644 what is jedi? Is that something the plug-in works with or something I can setup?

Thanks!

**Comment by levin-ilya** [comment]
Found the jedi python library, will try to setup that up, thanks for the tip!

**Comment by rampage644** [comment]
@levin-ilya: `jedi` is what python plugin uses behind the scenes for intellisense, afaik

**Comment by DonJayamanne** [comment]
Thanks for the research @rampage644

**Comment by i5on9i** [comment]
@DonJayamanne I've tested with just small project which has 7 .py files. At this, Go To definition, Symbols are working but with the big project, for example my django project, they do not work.

**Comment by tbarbugli** [comment]
`Go to definition` does nothing for me (followed the docs about virtualenv setup)

**Comment by DonJayamanne** [comment]
Is go to definition not working with the same file or across files?

**Comment by tbarbugli** [comment]
nothing works for me, same file / project / virtual env. not even stdlib functions are resolved. Is there any sort of log information available?

**Comment by DonJayamanne** [comment]
can you create a file with the following code.
Then try to go to the definition of "one" in line 3 and check whether that works.
```python
one = 1
two = 2
print(one + two)
```

**Comment by i5on9i** [comment]
@DonJayamanne 
I hope that my case can give you a hint. In my case, nothing works too. in the project which "find symbol" does not work, I made a .py file like you say, but the gotoDefinition function does not work. 

In my case, all the symbol related functions does not work and even in any small size of .py file in this project.

When I run "gotoSymbol" function, even in a small file, it does not shows anything, but loading sign.
"editor.action.referenceSearch.trigger" also does not work with showing only loading sign.

**Comment by DonJayamanne** [comment]
This is highly unusual, give me some time. I will prepare some patches with some logging.

**Comment by i5on9i** [comment]
these days, i opened the developer tools. I could see the below error when I opened the vscode with a python project.
```
error jediProxy 
Error (error) Error: spawn python ENOENT
```
IMO, that is why "Go to Symbol in File" does not work. Is there any way to resolve it?

Env.
Os : windows 10
vscode : 1.6.1
Python extension : 0.5.0

**Comment by i5on9i** [comment]
Now, I might get a solution.
After I set the python path on settings.json like the below. I could get rid of the error.
After setting it, I could not see the error I've mentioned
```
{
    "python.pythonPath": "c:/envs/python/torrent/Scripts/python.exe"
}
```

However, after I remove it, I could see the same error and I couldn't use the 'Go to Symbols in File'.
I thinks the problem that I'd had is because of the python path which may be used by the python extension.

I think it's better to notify users to set python path when load the python import for the first time.

Take care

**Comment by DonJayamanne** [comment]
@i5on9i  Will do

**Comment by i5on9i** [comment]
Cool~ thanks in advance

**Comment by shaypal5** [comment]
Hey!

Sorry to be bringing this up again, but I have the exact same issue.

OSX El Capitan 10.11.6, VS Code 1.8.1, Latest version of your Python extension (just removed and reinstalled). My general python path is set to the python of the only virtualenv I'm using (a conda env, in my case), **and** I've also set the workspace interpreter to it (btw, this selection menu shows all installed python virtual environments, so it seems to be working).

**Comment by i5on9i** [comment]
@shaypal5 I thinks the [comments of rampage644](https://github.com/DonJayamanne/pythonVSCode/issues/138#issuecomment-224859600) can help you understand your problem.

**Comment by shaypal5** [comment]
@i5on9i Hey, man. I read this whole thread through and through.

I don't get it. The issue's name is "Show All Symbols returns no results" and it is closed. Is the final resolution of this issue is that it cannot be addresses because of the inherent jedi limitation, and so that it is out of scope even as a feature request? If so, I think the issue should be closed with an accompanying clarifying message. 

Additionally, @DonJayamanne request to check if this feature works for some simple cases is confusing, implying that it's working for him, and so that this is a bug.

**Comment by i5on9i** [comment]
@shaypal5 I can't sure what was this issue was, but I think it was that the 'Show All Symbols' returns NOTHING. The issue you are saying is more like [this issue](https://github.com/DonJayamanne/pythonVSCode/issues/93). 

Take care.

**Comment by rampage644** [comment]
@shaypal5 As far as I know, @DonJayamanne fixed this issue with a help of `ctags`. So all previous intellisense works with a help of `jedi`, but this feature uses `ctags`.

First thing that comes to mind is to check if it's installed, then check if it actually builds symbols (and creates `ctags` file): try "Build Workspace Symbols" command. Also look for errors in "Developer tools" (Help - Toggle Developer Tools).

I can confirm it was wokring for me (until python3.6 broke everything).

**Comment by shaypal5** [comment]
@rampage644 : It was `ctags`! 🎊 

I just did `brew install ctags` and after restarting VS code everything works like a charm!

@DonJayamanne  I opened a pull request with a bullet point added to the Quick Start section of the README, mentioning the `jedi` and `ctags` dependencies.

@i5on9i  Btw, the issue I mentioned was actually exactly this issue; I even explicitly mentioned that Go To Symbol works flawlessly for me. :)
